### PR TITLE
Separate upsert/delete events for MASS (GSI 307)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Adds two new events for generalized resource upsertion and deletion: `searchable_resource_upserted` and `searchable_resource_deleted`.

Also adds one event for dataset deletion: `metadata_dataset_deleted`. The already existing event for dataset upsertion is left as `metadata_dataset_overview`.

### CODE REVIEW

**Review by Cito — APPROVED** [review]
Looks good.

File: ghga_event_schemas/pydantic_.py [hunk 1]
```diff
@@ -59,13 +59,19 @@ class MetadataDatasetID(BaseModel):
...
         ...,
         description="The name of the class this dataset/artifact resource corresponds to.",
     )
+
+
+class SearchableResource(SearchableResourceInfo):
+    """Model containing resource content in addition to the accession and class name"""
+
     content: dict[str, Any] = Field(
         ..., description="The metadata content of this dataset/artifact resource."
```
Comment (mephenor) [inline ghga_event_schemas/pydantic_.py:76]:
```suggestion
        ..., description="The metadata content of this artifact resource."
```

File: ghga_event_schemas/pydantic_.py [hunk 2]
```diff
@@ -59,13 +59,19 @@ class MetadataDatasetID(BaseModel):
...
 
-class MetadataDataset(MetadataDatasetID):
-    """Model to populate MASS from metldata artifacts"""
+class SearchableResourceInfo(BaseModel):
+    """Model containing only identifying information about an artifact's resource"""
 
+    accession: str = Field(..., description="The resource accession.")
     class_name: str = Field(
         ...,
         description="The name of the class this dataset/artifact resource corresponds to.",
```
Comment (mephenor) [inline ghga_event_schemas/pydantic_.py:68]:
```suggestion
        description="The name of the class this artifact resource corresponds to.",
```

**Review by mephenor — CHANGES_REQUESTED** [review]
Looks good
I'd just slightly change the descriptions to be in line with the more generic thing they are describing now
Probably also makes sense to remove "artiacts" from the MetadataDatasetID description to avoid confusion

**Review by mephenor — APPROVED** [review]

**Pull request merged**